_posts/en/newsletters/2020-09-30-newsletter.md

```diff
@@ -0,0 +1,184 @@
+---
+title: 'Bitcoin Optech Newsletter #117'
+permalink: /en/newsletters/2020/09/30/
+name: 2020-09-30-newsletter
+slug: 2020-09-30-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes a compiler bug that casts doubt on the
+safety of secure systems and explains a technique that can be used to more
+efficiently verify ECDSA signatures in Bitcoin. Also included are our regular
+sections with popular
+questions and answers from the Bitcoin StackExchange, announcements of
+releases and release candidates, and summaries of notable changes to
+popular Bitcoin infrastructure software.
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **Discussion about compiler bugs:** a [test][oconnor test] written
+  last week by Russell O'Connor for libsecp256k1 failed due to a
+  [bug][gcc bug] in the GNU Compiler Collection's (GCC's) built-in version of the
+  standard C library's `memcmp` (memory compare) function.  This function takes two
+  regions of memory, interprets them as integer values, and returns
+  whether the first region is less than, equal to, or greater than the
+  second region.  This is a commonly used low-level function.
+  Programs are almost never designed to verify such fundamental
+  operations were performed correctly, so bugs of this type can easily
+  result in a program producing incorrect results.  Those incorrect
+  results are possible even if the program's code was well reviewed,
+  strongly tested, formally verified, and otherwise built with the
+  utmost care.  A single incorrect result in the execution of
+  cryptographic or consensus code could result in serious consequences
+  for the users of that code---or anyone who depends on remaining in
+  consensus with users of that code.
+
+    The potential issues affect not only software written in C and
+    compiled with GCC, but any software or library that depends on
+    software or libraries compiled with the affected versions of GCC.
+    That may be a significant amount of software on affected systems,
+    such as recent versions of [Ubuntu and Fedora][maxwell systems].  It
```

Comment:
I think this could be toned done a little bit. The bug appeared scary at first glance but from all we know by now, it occurs only in very rare specific circumstances. This is confirmed by the fact that the bug has been known for over four months, and noone really made a fuzz about it on the GCC bugtracker.

Reply:
Just checking if you've left this unaddressed intentionally or if it has just been overlooked. This is certainly not a showstopper.

Reply:
Oops, I dropped this. Thanks for the reminder. Changed the wording to:

> The potential issues affect not only software written in C and compiled with GCC, but any software or library that depends on software or libraries compiled with the affected versions of GCC. It also includes programs written in other languages whose compilers or interpreters were built using GCC. Although the full extent of the issue is still unknown, Russell O’Connor has run an analysis of an entire Linux system and found only a handful of instances where this bug causes a miscompilation, suggesting that it’s fairly rare for code to be miscompiled due to the bug.